scipy/optimize/_brent.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import division
+
+from math import sqrt
+
+from numpy import finfo
+
+_sqrt_epsilon = sqrt(finfo(float).eps)
+# about 0.3819660112501051
+_golden = 0.5 * (3.0 - sqrt(5.0))
+
+
+def find_minimum(f, a, b, rtol=_sqrt_epsilon, atol=_sqrt_epsilon, maxiter=500):
```

Comment:
How does this behave if most points c between a and b have f(c)>f(a) and f(c)>f(b)? If, for example, the initial guess x0 is higher than either endpoint? Supplying a known minimum candidate c can ensure that the function doesn't settle on an endpoint.